src/index.js

```diff
@@ -63,41 +79,61 @@ class JSONTree extends React.Component {
   };
 
   static defaultProps = {
-    shouldExpandNode: (keyName, data, level) => level === 0, // expands root by default,
+    shouldExpandNode: expandRootNode,
     hideRoot: false,
     keyPath: ['root'],
-    getItemString: (type, data, itemType, itemString) => <Text>{itemType} {itemString}</Text>,
-    labelRenderer: ([label]) => <Text>{label}:</Text>,
+    getItemString: defaultItemString,
+    labelRenderer: defaultLabelRenderer,
     valueRenderer: identity,
     postprocessValue: identity,
-    isCustomNode: () => false,
+    isCustomNode: noCustomNode,
     collectionLimit: 50,
     invertTheme: true,
     sortObjectKeys: true,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = getStateFromProps(props);
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
It looks like this  `UNSAFE_componentWillReceiveProps` and the constructor could be replaced with a [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops)

Reply:
```
Warning: `JSONTree` uses `getDerivedStateFromProps` but its initial state is undefined. This is not recommended. Instead, define the initial state by assigning an object to `this.state` in the constructor of `JSONTree`. This ensures that `getDerivedStateFromProps` arguments have a consistent shape.
```
I think we should keep the constructor initial state setting.